src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -44,34 +48,40 @@ void setUserStore(UserStore userStore) {
   void setConversationStore(ConversationStore conversationStore) {
     this.conversationStore = conversationStore;
   }
+
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }
   /**
    * This function fires when a user requests the /profile URL. It simply forwards the request to
    * admin.jsp.
    */
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response)
-      throws IOException, ServletException {
-    request.getRequestDispatcher("/WEB-INF/view/admin.jsp").forward(request, response);
-    String username = (String) request.getSession().getAttribute("user");
-        if (username == null) {
-  //User is not logged in
-          response.sendRedirect("/login");
-          return;
-        }
+   public void doGet(HttpServletRequest request, HttpServletResponse response)
+        throws IOException, ServletException {
+      String username = (String) request.getSession().getAttribute("user");
+      boolean isRegistered = (username != null);
+      boolean isAdmin = isAdmin(username);
 
-    User user = userStore.getUser(username);
-        if (user == null) {
-  //User was not found
-          System.out.println("User not found: " + username);
-          response.sendRedirect("/login");
-          return;
-        }
-  //Users in this list can access the admin page
```

Comment:
For the preliminary purposes hard coding the users is ok, but think about how you can change this. This would e a really big security issue if you were to release this to the real world.

Reply:
i fixed this and tried a different approach

Reply:
i updated this with  a different approach